public/app/features/dashboard/utils/tracking.ts

```diff
@@ -0,0 +1,24 @@
+import { reportInteraction } from '@grafana/runtime';
+
+import { DashboardModel } from '../state';
+
+export function trackDashboardLoaded(dashboard: DashboardModel) {
+  // Count the different types of variables
+  const variables = dashboard.templating.list
+    .map((v) => v.type)
+    .reduce((r, k) => {
+      r[variableName(k)] = 1 + r[variableName(k)] || 1;
+      return r;
+    }, {});
+
+  reportInteraction('dashboard_loaded', {
+    uid: dashboard.uid,
+    title: dashboard.title,
+    style: dashboard.style,
+    schemaVersion: dashboard.schemaVersion,
+    panels_count: dashboard.panels.length,
+    ...variables,
```

Comment:
Do you know if Rudderstack supports nesting metadata? something like `{variables: {query: 1}}`

Reply:
I don't I'll try to find that out.

Reply:
Ok, I've done my research and I think this is not possible for the `properties` field. https://www.rudderstack.com/docs/api/http-api/#track Although it is possible to have a json field https://www.rudderstack.com/docs/destinations/warehouse-destinations/json-column-support/#using-the-json-column-feature but I'm not sure if we want to set that up.

Reply:
It is ok! We can namespace them under `variables`. Thank you for checking 😄